MultiSpeedSlider/Views/MultiSpeedSlider.swift

```diff
@@ -69,54 +69,54 @@ public class MultiSpeedSlider: UISlider {
          * back to the track after tracking in one of the slower tracking zones.
          */
-        let trackRect = trackRectForBounds(bounds)
-        let thumbRect = thumbRectForBounds(bounds, trackRect: trackRect, value: value)
-        let x = thumbRect.origin.x + (thumbRect.size.width/2)
-        let y = thumbRect.origin.y + (thumbRect.size.height/2)
-        beganTrackingLocation = CGPointMake(x, y)
+        let trackRect1 = trackRect(forBounds: bounds)
+        let thumbRect1 = thumbRect(forBounds: bounds, trackRect: trackRect1, value: value)
+        let x = thumbRect1.origin.x + (thumbRect1.size.width/2)
+        let y = thumbRect1.origin.y + (thumbRect1.size.height/2)
+        beganTrackingLocation = CGPoint(x: x, y: y)
         realPositionValue = value
     }
     
     //MARK: - Touch Events
-    public override func pointInside(point: CGPoint, withEvent event: UIEvent?) -> Bool {
-        let bounds = CGRectInset(self.bounds, -thumbTouchSize.width, -thumbTouchSize.height)
-        return CGRectContainsPoint(bounds, point)
+    public override func point(inside point: CGPoint, with event: UIEvent?) -> Bool {
+        let bounds = self.bounds.insetBy(dx: -thumbTouchSize.width, dy: -thumbTouchSize.height)
+        return bounds.contains(point)
     }
     
-    public override func beginTrackingWithTouch(touch: UITouch, withEvent event: UIEvent?) -> Bool {
-        let beginTracking = beginTrackWithTouchUsingExtendedTouchArea(touch) || allowTap
+    public override func beginTracking(_ touch: UITouch, with event: UIEvent?) -> Bool {
+        let beginTracking = beginTrackWithTouchUsingExtendedTouchArea(touch: touch) || allowTap
 
         if allowTap {
-            let tapPoint = touch.locationInView(self)
+            let tapPoint = touch.location(in: self)
             let valueFromTap = CGFloat(maximumValue - minimumValue) * (tapPoint.x - CGFloat(thumbTouchSize.width/2)) / (frame.size.width - thumbTouchSize.width)
             let newValue = minimumValue + Float(valueFromTap)
             setValue(newValue, animated: true)
-            sendActionsForControlEvents(.TouchDragInside)
-            sendActionsForControlEvents(.ValueChanged)
+            sendActions(for: .touchDragInside)
+            sendActions(for: .valueChanged)
         }
 
         if beginTracking {
             beganTracking()
-            delegate?.scrubbingStatusChanged(true)
+            delegate?.scrubbingStatusChanged(nowScrubbing: true)
         }
 
         return beginTracking
     }
     
-    public override func continueTrackingWithTouch(touch: UITouch, withEvent event: UIEvent?) -> Bool {
-        if tracking {
-            let previousLocation = touch.previousLocationInView(self)
-            let currentLocation = touch.locationInView(self)
+    public override func continueTracking(_ touch: UITouch, with event: UIEvent?) -> Bool {
+        if isTracking {
```

Comment:
all the code in this method can be indented one tab to the left if we change from `if isTracking {` to `guard isTracking else { return false  } in top and then remove indentation.`